cmd/juju/cloud/remove.go

```diff
@@ -102,33 +93,23 @@ func (c *removeCloudCommand) Init(args []string) (err error) {
 }
 
 func (c *removeCloudCommand) Run(ctxt *cmd.Context) error {
-	if c.BothClientAndController || c.ControllerOnly {
-		if c.ControllerName == "" {
-			// The user may have specified the controller via a --controller option.
-			// If not, let's see if there is a current controller that can be detected.
-			var err error
-			c.ControllerName, err = c.MaybePromptCurrentController(ctxt, fmt.Sprintf("remove cloud %v from", c.Cloud))
-			if err != nil {
-				return errors.Trace(err)
-			}
-		}
+	if err := c.MaybePrompt(ctxt, fmt.Sprintf("remove cloud %v from", c.Cloud)); err != nil {
+		return errors.Trace(err)
 	}
-	if c.ControllerName == "" && !c.ClientOnly {
-		ctxt.Infof("To remove cloud %q from this client, use the --client-only option.", c.Cloud)
+	if c.ControllerName == "" && !c.Client {
+		ctxt.Infof("To remove cloud %q from this client, use the --client option.", c.Cloud)
```

Comment:
We can return here right?
If not controller and not client then nothing to do.
Should this logic be used elsewhere? It seems this is the only place where we do this check and can exit early? Maybe I'm wrong, lots of code to look at

Reply:
Actually, I do not think that we need it anymore...

It looks like long ago, the original intent of this code was to detect that the controller and client operation was requested but no controller was specified/detected. In this case, in the next run of this command, the user could have just said "--client"... This is what this message is about - the information/education of the user. 

I do not think it's needed anymore as we no longer detect controllers and users need to specify them explicitly.